src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,81 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { actions as currentTodoActions } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(store => store.currentTodo);
+
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .catch((error) => {
+          throw error;
+        });
+    }
+  }, [currentTodo]);
+
+  const handleOnCloseModal = () => {
+    dispatch(currentTodoActions.removeTodo());
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {!user && <Loader />}
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+      {!!user && (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {`Todo #${currentTodo?.id}`}
+            </div>
+
+            <button
+              aria-label="Modal Close"
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={handleOnCloseModal}
+            />
+          </header>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {currentTodo?.title}
+            </p>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">fugiat veniam minus</p>
+            <p className="block" data-cy="modal-user">
+              {!currentTodo?.completed && (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              {currentTodo?.completed && (
+                <strong className="has-text-success">Done</strong>
+              )}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              {user && (
```

Comment:
instead of using a condition and a fragment it is better to write <a href={`mailto:${user?.email}`}>{user?.name}</a>

Reply:
> instead of using a condition and a fragment it is better to write <a href={`mailto:${user?.email}`}>{user?.name}

No, it's not better, in this case you could have invalid `href`